Explorer/Assets/DCL/Landscape/Utils/TerrainGeneratorLocalCache.cs

```diff
@@ -47,51 +47,102 @@ public class TerrainLocalCache
         private bool isValid;
         private string checksum;
         private const string FILE_NAME = "/terrain_cache";
-        private static readonly BinaryFormatter FORMATTER = new ();
+        private const string DICTIONARY_PATH = "/terrain_cache_dictionaries/";
+
+
+        public const string ALPHA_MAPS = "alphaMaps";
+        public const string HEIGHTS = "heights";
+        public const string TREES = "trees";
+        public const string DETAIL_LAYER = "detailLayer";
+        public const string HOLES = "holes";
+        
+        
+        public static readonly BinaryFormatter FORMATTER = new();
 
-        public Dictionary<int2, float[]> heights = new ();
         public int heightX;
         public int heightY;
 
-        public Dictionary<int2, float[]> alphaMaps = new ();
         public int alphaX;
         public int alphaY;
         public int alphaZ;
 
-        public Dictionary<int2, TreeInstanceDTO[]> trees = new ();
-
-        public Dictionary<int3, int[]> detail = new ();
         public int detailX;
         public int detailY;
 
-        public Dictionary<int2, bool[]> holes = new ();
         public int holesX;
         public int holesY;
 
         public int maxHeight;
 
         private TerrainLocalCache() { }
 
-        public void SaveToFile(int seed, int chunkSize, int version, string parcelChecksum)
+        public void SaveMetadataToFile(int seed, int chunkSize, int version, string parcelChecksum)
         {
-            string path = GetPath(seed, chunkSize, version);
+            var path = GetFilePath(seed, chunkSize, version);
             checksum = parcelChecksum;
 
             if (File.Exists(path))
+            {
                 File.Delete(path);
-
+                var dictionaryPath = GetDictionaryDirectory();
+                if (Directory.Exists(dictionaryPath))
+                    Directory.Delete(GetDictionaryDirectory(), true);
+            }
+            
             using FileStream fileStream = File.Create(path);
             FORMATTER.Serialize(fileStream, this);
+
         }
 
-        private static string GetPath(int seed, int chunkSize, int version) =>
-            Application.persistentDataPath + FILE_NAME + $"_{seed}_{chunkSize}_v{version}.data";
+        public void SaveArrayToFile<T>(string name, string offsetX, string offsetZ, T[] arrayToSave) where T : struct
+        {
+            var pathForDictionary = GetDictionaryFilePath(name, offsetX, offsetZ);
+            using var fileStreamForHeights = File.Create(pathForDictionary);
+            FORMATTER.Serialize(fileStreamForHeights, arrayToSave);
+        }
 
-        public static TerrainLocalCache NewEmpty() =>
-            new()
-            {
-                isValid = false
-            };
+        public void SaveArrayToFile<T>(string name, string offsetX, string offsetZ, string layer, T[] arrayToSave)
```

Comment:
Is this save functuion used? I think its duplicated?

Reply:
Its actually an overload, since the indexing is different